### PR TITLE
Prevent URL opener from overwriting existing repositories

### PR DESCRIPTION
**Pull request by kaeptmblaubaer1000**
This also fixes the issue that ports would be ignored in the opener (although they aren't necessary for the listed sites)

### CODE REVIEW

File: app/src/main/java/me/sheimi/sgit/RepoListActivity.java [hunk 1]
```diff
@@ -64,44 +64,45 @@ protected void onCreate(Bundle savedInstanceState) {
...
                     startActivity(intent);
-                }
-                else{
-					final String cloningStatus = getString(R.string.cloning);
-	                Repo mRepo = Repo.createRepo(repoName, repoUrlBuilder.toString(), cloningStatus );
+                } else if (Repo.getDir(((SGitApplication) getApplicationContext()).getPrefenceHelper(), repoName).exists()) {
+                    // Repository with name end already exists, see https://github.com/maks/MGit/issues/289
+                    CloneDialog cloneDialog = new CloneDialog();
+                    cloneDialog.setUrl(repoUrlBuilder.toString());
+                    cloneDialog.show(getSupportFragmentManager(), "clone-dialog");
```
[the remark this answers is not in the document]
Comment (maks) [reply app/src/main/java/me/sheimi/sgit/RepoListActivity.java]: Ok I see now, you prefill the clone dialog with the url and then let the normal flow take place which will alert the user to an already existing repo.

File: app/src/main/java/me/sheimi/sgit/dialogs/CloneDialog.java [hunk 2]
```diff
@@ -165,4 +170,12 @@ public void cloneRepo() {
         CloneTask task = new CloneTask(mRepo, mBinding.cloneRecursive.isChecked(), getString(R.string.cloning), null);
         task.executeTask();
     }
+
+    public void setUrl(String url) {
+        if (mBinding == null) {
```
Comment (maks) [inline app/src/main/java/me/sheimi/sgit/dialogs/CloneDialog.java]: This is not really the way to make use of databinding. Instead you want to set a variable in the dialog that is then used within the layout resource via databinding.

Comment (kaeptmblaubaer1000) [reply app/src/main/java/me/sheimi/sgit/dialogs/CloneDialog.java]: @maks how would you do this?
I haven't ever used Databinding.

**Comment by kaeptmblaubaer1000** [comment]
@maks the initialization issue still throws NPE.

**Pull request merged**